docs/web3-eth-admin.rst

```diff
@@ -0,0 +1,423 @@
+    .then(console.log);
+    > true
+
+------------------------------------------------------------------------------
+
+
+getDataDirectory
+=====================
+
+.. code-block:: javascript
+
+    web3.eth.admin.getDataDirectory([, callback])
+
+Provides absolute path of the running node, which is used by the node to store all its databases.
+
+----------
+Parameters
+----------
+
+
+1. ``Function`` - (optional) Optional callback, returns an error object as first parameter and the result as second.
+
+
+-------
+Returns
+-------
+
+
+``Promise<string>`` - The path.
+
+
+-------
+Example
+-------
+
+
+.. code-block:: javascript
+
+    web3.eth.admin.getDataDirectory()
+    .then(console.log);
+    > "/home/ubuntu/.ethereum"
+
+
+------------------------------------------------------------------------------
+
+
+getNodeInfo
+=====================
+
+.. code-block:: javascript
+
+    web3.eth.personal.getNodeInfo([, callback])
+
+This property can be queried for all the information known about the running node at the networking granularity..
+
+----------
+Parameters
+----------
+
+
+1. ``Function`` - (optional) Optional callback, returns an error object as first parameter and the result as second.
+
+
+-------
+Returns
+-------
+
+
+``Promise<object>`` - The node information array.
+
+
+-------
+Example
+-------
+
+
+.. code-block:: javascript
+
+    web3.eth.admin.getNodeInfo().then(console.log);
+    > {
+        enode: "enode://44826a5d6a55f88a18298bca4773fca5749cdc3a5c9f308aa7d810e9b31123f3e7c5fba0b1d70aac5308426f47df2a128a6747040a3815cc7dd7167d03be320d@[::]:30303",
+        id: "44826a5d6a55f88a18298bca4773fca5749cdc3a5c9f308aa7d810e9b31123f3e7c5fba0b1d70aac5308426f47df2a128a6747040a3815cc7dd7167d03be320d",
+        ip: "::",
+        listenAddr: "[::]:30303",
+        name: "Geth/v1.5.0-unstable/linux/go1.6",
+        ports: {
+            discovery: 30303,
+            listener: 30303
+        },
+        protocols: {
+            eth: {
+            difficulty: 17334254859343145000,
+            genesis: "0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3",
+            head: "0xb83f73fbe6220c111136aefd27b160bf4a34085c65ba89f24246b3162257c36a",
+            network: 1
+            }
+        }
+    }
+
+------------------------------------------------------------------------------
+
+
+getPeers
+=====================
+
+.. code-block:: javascript
+
+    web3.eth.admin.getPeers([, callback])
+
+This will provide all the information known about the connected remote nodes at the networking granularity.
+
+----------
+Parameters
+----------
+
+
+1. ``Function`` - (optional) Optional callback, returns an error object as first parameter and the result as second.
+
+
+-------
+Returns
+-------
+
+
+``Promise<ArrayBuffer>`` - List of all connected peers.
```

Comment:
why is this a type of `ArrayBuffer` can it grow in size after it has been initialised?